### PR TITLE
refactor aaq to use modelforms

### PR DESCRIPTION
**Pull request by LeoMcA**

### CODE REVIEW

File: kitsune/questions/forms.py [hunk 1]
```diff
@@ -315,36 +299,13 @@ def __init__(self, product=None, *args, **kwargs):
...
-
-        question = Question(
-            creator=user,
-            title=self.cleaned_data["title"],
-            content=self.cleaned_data["content"],
-            locale=locale,
-            product=product
-        )
+    def save(self, user, locale, product, product_config, *args, **kwargs):
+        question = super(NewQuestionForm, self).save(commit=False, *args, **kwargs)
```
Comment (akatsoulas) [inline kitsune/questions/forms.py:305]: Why are you using `commit=False` here?

File: kitsune/questions/views.py [hunk 2]
```diff
@@ -556,82 +573,64 @@ def aaq(
...
-                "topics": topics
-            },
+    form = None
+    images = None
+    featured = []
+    topics = []
+
+    if step == 2:
+        featured = get_featured_articles(product)
+        topics = topics_for(product, parent=None)
```
Comment (akatsoulas) [inline kitsune/questions/views.py:583]: Why don't you query here the db to get the topics for this product?

File: kitsune/questions/forms.py (hunk 1 above)
Comment (LeoMcA) [reply kitsune/questions/forms.py:305]: Otherwise it errors because the creator_id column is blank

File: kitsune/questions/views.py (hunk 2 above)
Comment (LeoMcA) [reply kitsune/questions/views.py:583]: it seemed easier to use this helper function

File: kitsune/questions/forms.py (hunk 1 above)
Comment (akatsoulas) [reply kitsune/questions/forms.py:305]: Wouldn't something like this work:
```python
self.instance.creator = user
super(NewQuestionForm, self).save()
```

Comment (LeoMcA) [reply kitsune/questions/forms.py:305]: That's clever, update incoming

File: kitsune/questions/views.py [hunk 3]
```diff
@@ -556,93 +573,72 @@ def aaq(
...
         request,
         template,
-        {
-            "form": form,
-            "images": images,
-            "products": config.products,
-            "current_product": product_config,
-            "current_step": step,
-        },
+        context
```
Comment (akatsoulas) [inline kitsune/questions/views.py:641]: this fits now in a single line

**Pull request merged**